### PR TITLE
fix(vllm): resolve PD connector from MultiConnector config

### PR DESCRIPTION
**Pull request by Moersity**
### Problem

`GetServerInfo` currently reports the outer `MultiConnector` name and engine ID verbatim. The router only recognizes concrete PD connectors, so a worker whose `MultiConnector` wraps one `MooncakeConnector` or `NixlConnector` loses PD capability discovery even though its configuration is valid.

### Solution

Resolve the effective PD connector before building `GetServerInfoResponse`. A `MultiConnector` is projected only when its child list is structurally valid, contains exactly one supported PD connector, and resolves to a non-empty engine ID. All malformed, store-only, empty-ID, or ambiguous configurations continue to report `MultiConnector` so discovery fails closed instead of guessing.

The child `engine_id` takes precedence when present; otherwise the parent ID is used, matching vLLM's own child `KVTransferConfig` construction.

## Changes

- Add a small engine-independent `resolve_pd_connector` helper in the vLLM KV-transfer module.
- Use the resolved connector and engine ID in `GetServerInfo` while preserving the existing role and ordinary-connector behavior.
- Add engine-free regression tests for Mooncake, NIXL, parent/child engine IDs, empty engine IDs, malformed children, store-only wrappers, ambiguous PD children, and ordinary connectors.

## Test Plan

Executed on the project's CI host:

- Before the empty-ID fix, `PYTHONDONTWRITEBYTECODE=1 /tmp/smg-pr2425-review-venv/bin/python -m pytest -q -p no:cacheprovider grpc_servicer/tests/test_kv_transfer_params.py::TestResolvePdConnector::test_empty_child_engine_id_is_not_projected grpc_servicer/tests/test_kv_transfer_params.py::TestResolvePdConnector::test_empty_parent_engine_id_is_not_projected` — 2 failed as expected; the child case returned `("NixlConnector", "")`, and the inherited-parent case returned `("MooncakeConnector", "")`.
- The same two-node command after the fix — 2 passed.
- `PYTHONDONTWRITEBYTECODE=1 /tmp/smg-pr2425-review-venv/bin/python -m pytest -q -p no:cacheprovider grpc_servicer/tests/test_kv_transfer_params.py::TestResolvePdConnector` — 14 passed.
- `PYTHONDONTWRITEBYTECODE=1 /tmp/smg-pr2425-review-venv/bin/python -m pytest -q -p no:cacheprovider grpc_servicer/tests/test_kv_transfer_params.py` — 34 passed.
- `PYTHONDONTWRITEBYTECODE=1 /tmp/smg-pr2425-review-venv/bin/python -m pytest -q -p no:cacheprovider grpc_servicer/tests` — 102 passed, 5 skipped.
- `PYTHONDONTWRITEBYTECODE=1 /tmp/smg-pr2425-review-venv/bin/python -m py_compile grpc_servicer/smg_grpc_servicer/vllm/kv_transfer.py grpc_servicer/smg_grpc_servicer/vllm/servicer.py grpc_servicer/tests/test_kv_transfer_params.py` — passed.
- `/tmp/smg-pr2425-review-venv/bin/ruff format --check grpc_servicer/smg_grpc_servicer/vllm/kv_transfer.py grpc_servicer/smg_grpc_servicer/vllm/servicer.py grpc_servicer/tests/test_kv_transfer_params.py` — 3 files already formatted.
- `/tmp/smg-pr2425-review-venv/bin/ruff check grpc_servicer/smg_grpc_servicer/vllm/kv_transfer.py grpc_servicer/smg_grpc_servicer/vllm/servicer.py grpc_servicer/tests/test_kv_transfer_params.py` — all checks passed.

## Compatibility and risk

SMG supports vLLM 0.19.0 and newer and currently tests 0.27.1. Both versions represent `MultiConnector` children as a list of `KVTransferConfig` dictionaries under `kv_connector_extra_config["connectors"]`, and both inherit the parent engine ID when a child omits it. The helper depends only on that stable public configuration shape and does not import vLLM.

Risk is limited to server-info projection for `MultiConnector`. Ordinary connectors are returned unchanged, and any unrecognized, empty-ID, or ambiguous wrapper remains a `MultiConnector`. No protobuf or routing contracts change.

<details>
<summary>Checklist</summary>

- [x] Changed Python files pass format and lint checks
- [x] Regression tests cover successful and fail-closed resolution
- [x] Cargo checks are not applicable because no Rust files changed
- [x] Documentation is not required for this internal discovery correction

</details>

### CODE REVIEW

File: grpc_servicer/smg_grpc_servicer/vllm/kv_transfer.py [hunk 1]
```diff
@@ -7,6 +7,48 @@
...
+            return connector, engine_id
+        if child_connector in _SUPPORTED_PD_CONNECTORS:
+            pd_children.append(child)
+
+    if len(pd_children) != 1:
+        return connector, engine_id
+
+    child = pd_children[0]
+    child_engine_id = child.get("engine_id", engine_id)
+    if not isinstance(child_engine_id, str):
```
[the remark this answers is not in the document]
Comment (Moersity) [reply grpc_servicer/smg_grpc_servicer/vllm/kv_transfer.py]: Fixed in 0cd3f993 — reject empty effective engine IDs before projection and cover both explicit child-empty and inherited parent-empty cases.
